service/lib/agama/storage/iscsi/manager.rb

```diff
@@ -296,26 +297,51 @@ def apply_initiator_config(config)
         def apply_targets_config(config)
           return unless config.targets
 
-          start_progress_with_size(2)
+          start_progress_with_size(3)
           progress.step(_("Logout iSCSI targets")) { logout_targets }
-          progress.step(_("Login iSCSI targets")) { login_targets(config.targets) }
+          progress.step(_("Discover iSCSI targets")) { discover_from_portals(config) }
+          progress.step(_("Login iSCSI targets")) { login_targets(config) }
         end
 
         # Tries to logout from all targets (nodes).
         def logout_targets
           nodes.select(&:connected?).each { |n| adapter.logout(n) }
         end
 
-        # Tries to login to all given targets.
+        # Discovers iSCSI targets from all the portals.
+        #
+        # @param config [ISCSI::Config]
+        def discover_from_portals(config)
+          config.portals.each { |p| discover_from_portal(p, interfaces: config.interfaces(p)) }
+        end
+
+        # Discovers iSCSI targets from a portal.
+        #
+        # @param portal [String]
+        # @param interfaces [Array<String>]
+        def discover_from_portal(portal, interfaces: [])
+          iface_params = interfaces.map { |i| ["-I", i] }.flatten
+
+          Yast::Execute.locally!(
```

Comment:
I didn't expect a call to `iscsiadm` here. At most, I would expect something like:

- `IscsiClientLib` calls `iscsiadm`
- `Agama::Storage::ISCSI::Adapter` relies on `IscsiClientLib`
- The iSCSI manager relies on  the adapter.

Maybe some shorcuts are needed for this, but calling `iscsiadm` directly from the manager (which is the upper level of the current architecture), looks like quite a leap to me.